adapters/searchpress.php

```diff
@@ -1,77 +1,117 @@
-		'term_name'             => 'terms.%s.name.raw',
-		'category_name'         => 'terms.%s.name.raw',
-		'tag_name'              => 'terms.%s.name.raw',
-	), $es_map );
+	return wp_parse_args(
+		array(
+			'post_name'             => 'post_name.raw',
+			'post_title'            => 'post_title.raw',
+			'post_title.analyzed'   => 'post_title',
+			'post_content.analyzed' => 'post_content',
+			'post_author'           => 'post_author.user_id',
+			'post_date'             => 'post_date.date',
+			'post_date_gmt'         => 'post_date_gmt.date',
+			'post_modified'         => 'post_modified.date',
+			'post_modified_gmt'     => 'post_modified_gmt.date',
+			'post_type'             => 'post_type.raw',
+			'post_meta'             => 'post_meta.%s.raw',
+			'post_meta.analyzed'    => 'post_meta.%s.value',
+			'post_meta.signed'      => 'post_meta.%s.long',
+			'post_meta.unsigned'    => 'post_meta.%s.long',
+			'term_name'             => 'terms.%s.name.raw',
+			'category_name'         => 'terms.%s.name.raw',
+			'tag_name'              => 'terms.%s.name.raw',
+		),
+		$es_map 
+	);
 }
 add_filter( 'es_field_map', 'sp_es_field_map' );
 
-
+// This section only used for unit tests.
+// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped, WordPress.PHP.DevelopmentFunctions.error_log_print_r
 if ( defined( 'ES_WP_QUERY_TEST_ENV' ) && ES_WP_QUERY_TEST_ENV ) {
 
-	remove_action( 'save_post',       array( SP_Sync_Manager(), 'sync_post' ) );
-	remove_action( 'delete_post',     array( SP_Sync_Manager(), 'delete_post' ) );
-	remove_action( 'trashed_post',    array( SP_Sync_Manager(), 'delete_post' ) );
+	remove_action( 'save_post', array( SP_Sync_Manager(), 'sync_post' ) );
+	remove_action( 'delete_post', array( SP_Sync_Manager(), 'delete_post' ) );
+	remove_action( 'trashed_post', array( SP_Sync_Manager(), 'delete_post' ) );
 
+	/**
+	 * A function to make test data available in the index.
+	 */
 	function es_wp_query_index_test_data() {
 		// If your ES server is not at localhost:9200, you need to set $_ENV['searchpress_host'].
 		$host = ! empty( $_ENV['searchpress_host'] ) ? $_ENV['searchpress_host'] : 'http://localhost:9200';
 
-		SP_Config()->update_settings( array( 'active' => false, 'host' => $host ) );
+		SP_Config()->update_settings(
+			array(
+				'active' => false,
+				'host'   => $host,
+			) 
+		);
 		SP_API()->index = 'es-wp-query-tests';
 
 		SP_Config()->flush();
 		SP_Config()->create_mapping();
 
-		$posts = get_posts( array(
-			'posts_per_page' => -1,
-			'post_type' => 'any',
-			'post_status' => array_values( get_post_stati() ),
-			'orderby' => 'ID',
-			'order' => 'ASC',
-		) );
+		$posts = get_posts( // phpcs:ignore WordPressVIPMinimum.VIP.RestrictedFunctions.get_posts_get_posts
+			array(
+				'posts_per_page' => -1, // phpcs:ignore WordPress.VIP.PostsPerPage.posts_per_page_posts_per_page
+				'post_type'      => 'any',
+				'post_status'    => array_values( get_post_stati() ),
```

Comment:
TIL `stati`

Reply:
Don't put too much stock in the vocabulary used in WordPress code, [the plural of status isn't really stati](https://english.stackexchange.com/questions/877/what-is-the-plural-form-of-status).